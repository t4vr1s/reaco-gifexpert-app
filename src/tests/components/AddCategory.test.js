import '@testing-library/jest-dom'
import React from 'react'
import { shallow } from 'enzyme'
import { AddCategory } from '../../components/AddCategory'

/* eslint-disable */
describe('Pruebas en <AddCategory>', () => {
  const setCategories = jest.fn()
  let wrapper = shallow(<AddCategory setCategories={setCategories} />)

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories} />)
  })

  test('should de mostrar correctamente el componente', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should cambiar el valor del input', () => {
    const input = wrapper.find('input')
    const value = 'Rick and Morty'
    input.simulate('change', { target: { value } })
    expect(wrapper.find('p').text().trim()).toBe(value)
  })

  test('no debe enviar la informaciÃ³n con submit ', () => {
    wrapper.find('form').simulate('submit', { preventDefault: () => { } })
    expect(setCategories).not.toHaveBeenCalled()
  })

  test('debe llamar el setCategories y limpiar la caja de texto', () => {
    const input = wrapper.find('input')
    const value = 'Rick and Morty'
    input.simulate('change', { target: { value } })
    wrapper.find('form').simulate('submit', { preventDefault: () => { } })
    expect(setCategories).toHaveBeenCalled()
    expect(setCategories).toHaveBeenCalledTimes(1)
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function))

    expect(wrapper.find('input').text().trim()).toBe('')
    expect(wrapper.find('input').prop('value')).toBe('')
  })
})
