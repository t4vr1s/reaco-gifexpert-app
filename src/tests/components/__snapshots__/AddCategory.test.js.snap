// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory> should de mostrar correctamente el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
